sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -233,6 +241,10 @@ def lambda_handler(event, context):
 
     # send the paylaod to the opensearch instance
     client.send_payload(document_payload)
+
+    # take OpenSearch Snapshot
+    run_backup(host, region, snapshot_repo_name, snapshot_s3_bucket, snapshot_role_arn)
```

Comment:
So does this run a backup every single time a new file is added to the SDC?

Reply:
Yeah, each time the indexer is called it creates a new snapshot. Snapshots only capture the difference between what exists at the present and the previous snapshot, so more frequent snapshots will take less time to run and won't take up much more space as infrequent snapshots.

Reply:
Ah that's interesting! So when you restore, it automatically figures out how to aggregate the million snapshots? 

Do you know what would happen if we received like 100 files at once and opensearch was told to backup 100 times in rapid succession? Does it just listen to one at a time and queue up the others? I'm just curious if opensearch would get stressed out from all the requests or if it does some magic on the backend.